client/lib/domains/email-forwarding/index.js

```diff
@@ -7,4 +13,29 @@ function emailForwardingPlanLimit( plan ) {
 	return ( isBusiness( plan ) ? 100 : 5 );
 }
 
-module.exports = { emailForwardingPlanLimit };
+function validateAllFields( fieldValues ) {
+	return mapValues( fieldValues, ( value, fieldName ) => {
+		const isValid = validateField( {
+			value,
+			name: fieldName
+		} );
+
+		return isValid ? [] : [ 'Invalid' ];
+	} );
+}
+
+function validateField( { name, value } ) {
+	switch ( name ) {
+		case 'mailbox':
+			return value.length > 0 && /^([a-z0-9\._\+\-])*$/i.test( value ) && ! /(^\.)|(\.{2,})|(\.$)/.test( value );
+		case 'destination':
+			return value.length > 0 && emailValidator.validate( value );
```

Comment:
Looking at `email-validator`'s source code, it already checks for this :) And I think it's safe to assume that a good validator should check this for us.